src/context_flags.ml

```diff
@@ -0,0 +1,22 @@
+#directory "+compiler-libs";;
+#load "ocamlcommon.cma";;
+
+match Sys.argv.(1) with
+| "cxxflags" ->
+    if Sys.win32 && Config.ccomp_type = "msvc" then
+      print_string "(/EHsc)"
+    else
+      print_string "()"
+| "clibs" ->
+    if Config.ccomp_type = "cc" then
+      print_string "(-lstdc++)"
+    else
+      print_string "()"
+| "flags" ->
+    if Config.system = "mingw" then
+      print_string "(-ccopt \"-link -shared-libgcc\")"
```

Comment:
mingw64 uses Windows Structured Exception Handler (SEH) which is a kernel feature. mingw32 still uses SJLJ exceptions, which requires shared libgcc to work (otherwise exceptions can't cross library boundaries)